src/lib/Scenes/Home/Components/ForYou/index.tsx

```diff
@@ -66,11 +66,7 @@ export class ForYou extends React.Component<ViewProperties & RelayProps, State>
     this.setState({ isRefreshing: true })
 
     try {
-      await Promise.all(
-        this.railModules.map(railModule => {
-          railModule.refreshData()
-        })
-      )
+      await Promise.all(this.railModules.map(railModule => railModule && railModule.refreshData()))
```

Comment:
Only refresh mounted rails.

Reply:
Ah, right on, nice and clear. Should we explicitly `Promise.reject()` here or is a falsy return ok for `Promise.all`? (I'm not totally certain)

Reply:
`Promise.all` just returns an array with return values from the promises (which is `undefined` for mounted components and `null` for unmounted components`), it doesn’t care about those values itself, so we’re good :+1: